eclipse-theia/theia
# Fix webpack build issues in production mode

There is a bug with a new dependency from `webpack`: Using
`webpack-sources@3` produces bogus bundles, failing our application
build process.

We will need to update the `webpack` range again once this issue is
fixed in whatever package is causing this build issue on our end.



#### What it does

Fixes https://github.com/eclipse-theia/theia/issues/9795 by constraining our `webpack` range to avoid the bogus version.

See https://github.com/webpack/webpack/issues/13887 for failure details.

Once this issue will be fixed by `webpack` and/or `webpack-sources` we should be able to revert the constrained range to something more "open" (like before).

#### How to test

To see the failure:

- Checkout master
- Run `yarn upgrade`
- Run `yarn`
- Run `yarn --cwd examples/browser theia build`

This should lead to an error coming out of `webpack`/`Terser`.

To make sure this issue is fixed:

- Checkout this PR
- Run `yarn`
- Run `yarn --cwd examples/browser theia build`

`yarn upgrade` is already part of those changes.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
